subsys/bluetooth/controller/hci/hci.c

```diff
@@ -744,6 +751,239 @@ static void read_bd_addr(struct net_buf *buf, struct net_buf **evt)
+	uint8_t i;
+
+	/* read standard codec information */
+	num_std_codecs = hci_vendor_read_std_codecs(&std_codecs);
+	std_codecs_bytes = num_std_codecs *
+				sizeof(struct bt_hci_std_codec_info_v2);
+	/* read vendor-specific codec information */
+	num_vs_codecs = hci_vendor_read_vs_codecs(&vs_codecs);
+	vs_codecs_bytes = num_vs_codecs *
+				sizeof(struct bt_hci_vs_codec_info_v2);
+
+	/* allocate response packet */
+	rp = hci_cmd_complete(evt, sizeof(*rp) +
+			      std_codecs_bytes +
+			      vs_codecs_bytes);
+	rp->status = 0x00;
+	ptr = (uint8_t *)&rp->status + sizeof(rp->status);
+
+	/* copy standard codec information */
+	*ptr++ = num_std_codecs;
+	for (i = 0; i < num_std_codecs; i++) {
+		struct bt_hci_std_codec_info_v2 codec;
+
+		codec.codec_id = std_codecs[i].codec_id;
+		codec.transport = std_codecs[i].transport;
+		memcpy(ptr, &codec, sizeof(codec));
+		ptr += sizeof(codec);
+	}
+
+	/* copy vendor specific codec information  */
+	*ptr++ = num_vs_codecs;
+	for (i = 0; i < num_std_codecs; i++) {
+		struct bt_hci_vs_codec_info_v2 codec;
+
+		codec.company_id = sys_cpu_to_le16(vs_codecs[i].company_id);
+		codec.codec_id = sys_cpu_to_le16(vs_codecs[i].codec_id);
+		codec.transport = vs_codecs[i].transport;
+		memcpy(ptr, &codec, sizeof(codec));
+		ptr += sizeof(codec);
+	}
+}
+
+uint8_t __weak hci_vendor_read_codec_capabs(uint8_t coding_format,
+					    uint16_t company_id,
+					    uint16_t vs_codec_id,
+					    uint8_t ltt,
+					    uint8_t dir,
+					    uint8_t *num_capabs,
+					    size_t *capabs_bytes,
+					    const uint8_t **capabs)
+{
+	ARG_UNUSED(coding_format);
+	ARG_UNUSED(company_id);
+	ARG_UNUSED(vs_codec_id);
+	ARG_UNUSED(ltt);
+	ARG_UNUSED(dir);
+	ARG_UNUSED(capabs);
+
+	*num_capabs = 0;
+	*capabs_bytes = 0;
+
+	/* return status */
+	return 0x00;
+}
+
+static void read_codec_capabs(struct net_buf *buf, struct net_buf **evt)
+{
+	struct bt_hci_cp_read_codec_capabs *cmd = (void *)buf->data;
+	struct bt_hci_rp_read_codec_capabs *rp;
+	const uint8_t *capabs;
+	uint16_t vs_codec_id;
+	uint16_t company_id;
+	size_t capabs_bytes;
+	uint8_t num_capabs;
+	uint8_t status;
+	uint8_t *ptr;
+
+	company_id = sys_le16_to_cpu(cmd->codec_id.company_id);
+	vs_codec_id = sys_le16_to_cpu(cmd->codec_id.vs_codec_id);
+
+	/* read codec capabilities */
+	status = hci_vendor_read_codec_capabs(cmd->codec_id.coding_format,
+					      company_id,
+					      vs_codec_id,
+					      cmd->ltt,
+					      cmd->dir,
+					      &num_capabs,
+					      &capabs_bytes,
+					      &capabs);
+
+	/* allocate response packet */
+	rp = hci_cmd_complete(evt, sizeof(*rp) + capabs_bytes);
+	rp->status = status;
+
+	/* copy codec capabilities information */
+	rp->num_capabs = num_capabs;
+	ptr = (uint8_t *)&rp->num_capabs + sizeof(rp->num_capabs);
+	memcpy(ptr, capabs, capabs_bytes);
+}
+
+uint8_t __weak hci_vendor_read_ctlr_delay(uint8_t coding_format,
+					  uint16_t company_id,
+					  uint16_t vs_codec_id,
+					  uint8_t ltt,
+					  uint8_t dir,
```

Comment:
```suggestion
					  uint8_t transport_type,
					  uint8_t direction,
```

Reply:
Updated to use `transport` and `direction`